lib/librarian/puppet/simple/installer.rb

```diff
@@ -53,7 +53,10 @@ def install_git(module_path, module_name, repo, ref = nil)
             system_cmd("git clone #{repo} #{module_name}")
             Dir.chdir(module_dir) do
               system_cmd('git branch -r')
-              system_cmd("git checkout #{ref}") if ref
+              if ref
+                system_cmd("git fetch origin #{ref}")
```

Comment:
I also _really_ don't like how we're hard coding a remote named `origin` here, but this is something that needs to be improved in a separate patch.